src/common/components/signInButton/signInButton.component.js

```diff
@@ -0,0 +1,66 @@
+import angular from 'angular';
+import 'angular-gettext';
+import includes from 'lodash/includes';
+import sessionService from 'common/services/session/session.service';
+import sessionModalService from 'common/services/session/sessionModal.service';
+
+import template from './signInButton.tpl';
+
+let componentName = 'signInButton';
+
+class SignInButtonController {
+
+  /* @ngInject */
+  constructor( $window, sessionService, sessionModalService ) {
+    this.$window = $window;
+    this.sessionService = sessionService;
+    this.sessionModalService = sessionModalService;
+  }
+
+  $onInit() {
+    this.subscription = this.sessionService.sessionSubject.subscribe( () => this.sessionChanged() );
+  }
+
+  $onDestroy() {
+    this.subscription.unsubscribe();
+  }
+
+  signIn( reload ) {
+    reload = angular.isUndefined( reload ) ? true : reload;
+    var signInPromise = this.sessionModalService.signIn();
+    if ( reload ) {
+      signInPromise.then( () => {
+        this.$window.location.reload();
+      } );
+    }
+  }
+
+  signOut( reload ) {
+    reload = angular.isUndefined( reload ) ? true : reload;
+    var signOutPromise = this.sessionService.signOut();
+    if ( reload ) {
+      signOutPromise.then( () => {
+        this.$window.location.reload();
+      } );
+    }
+  }
+
+  sessionChanged() {
+    this.isSignedIn = includes( ['IDENTIFIED', 'REGISTERED'], this.sessionService.getRole() );
+  }
+}
+
+export default angular
+  .module( componentName, [
+    'gettext',
+    template.name,
+    sessionService.name,
+    sessionModalService.name
+  ] )
+  .component( componentName, {
+    controller:  SignInButtonController,
+    templateUrl: template.name,
+    bindings:    {
+      pageReload: '<'
```

Comment:
Is this going to be coming in from a variable in some outer scope or just hardcoded as a string on each btn instance such as `<sign-in-button page-reload="false">`? If it's the second case, I think using the `@` binding might be better.